lib/ruby_lsp/requests/folding_ranges.rb

```diff
@@ -18,276 +18,154 @@ module Requests
+            start_line: chunk.first.location.start_line - 1,
+            end_line: chunk.last.location.end_line - 1,
+            kind: "comment",
+          )
         end
-
-        super
-      end
-
-      # This is to prevent duplicate ranges
-      sig { params(node: T.any(SyntaxTree::Command, SyntaxTree::CommandCall)).returns(T::Boolean) }
-      def same_lines_for_command_and_block?(node)
-        node_block = node.block
-        return false unless node_block
-
-        location = node.location
-        block_location = node_block.location
-        block_location.start_line == location.start_line && block_location.end_line == location.end_line
       end
 
-      class PartialRange
-        extend T::Sig
-
-        sig { returns(String) }
-        attr_reader :kind
-
-        sig { returns(Integer) }
-        attr_reader :end_line
-
-        class << self
-          extend T::Sig
-
-          sig { params(node: SyntaxTree::Node, kind: String).returns(PartialRange) }
-          def from(node, kind)
-            new(node.location.start_line - 1, node.location.end_line - 1, kind)
-          end
-        end
-
-        sig { params(start_line: Integer, end_line: Integer, kind: String).void }
-        def initialize(start_line, end_line, kind)
-          @start_line = start_line
-          @end_line = end_line
-          @kind = kind
-        end
-
-        sig { params(node: SyntaxTree::Node).returns(PartialRange) }
-        def extend_to(node)
-          @end_line = node.location.end_line - 1
-          self
-        end
-
-        sig { params(node: SyntaxTree::Node).returns(T::Boolean) }
-        def new_section?(node)
-          node.is_a?(SyntaxTree::Comment) && @end_line + 1 != node.location.start_line - 1
-        end
-
-        sig { returns(Interface::FoldingRange) }
-        def to_range
-          Interface::FoldingRange.new(
-            start_line: @start_line,
-            end_line: @end_line,
-            kind: @kind,
+      sig { void }
+      def emit_requires_range
+        if @requires.length > 1
+          @ranges << Interface::FoldingRange.new(
+            start_line: T.must(@requires.first).location.start_line - 1,
+            end_line: T.must(@requires.last).location.end_line - 1,
+            kind: "imports",
           )
         end
 
-        sig { returns(T::Boolean) }
-        def multiline?
-          @end_line > @start_line
-        end
+        @requires.clear
       end
 
-      sig { params(node: T.nilable(SyntaxTree::Node)).returns(T::Boolean) }
-      def handle_partial_range(node)
-        kind = partial_range_kind(node)
+      sig { override.params(node: T.nilable(YARP::Node)).void }
+      def visit(node)
+        emit_requires_range unless node.is_a?(YARP::CallNode)
 
-        if kind.nil?
-          emit_partial_range
-          return true
-        end
+        case node
+        when YARP::ArrayNode, YARP::BlockNode, YARP::CaseNode, YARP::ClassNode, YARP::ForNode, YARP::HashNode,
```

Comment:
I got rid of the constants listing all nodes because that makes it impossible for Sorbet to understand and help with typing due to the splat operator.

Putting them directly here means we get correct type and need fewer annotations.

Reply:
The new node in YARP for heredocs is the `InterpolatedStringNode`. It needs to be handled in its own branch because in YARP the location of the heredoc is not based on the opening and closing delimiters.